chicago/permit_cleaning.py

```diff
@@ -84,6 +84,13 @@ def expand_multi_pin_permits(df):
     # keep rows with NA for pin1, filter out rows with NA for other pins
     melted_df = melted_df[(melted_df["pin_type"] == "pin1") | ((melted_df["pin_type"] != "pin1") & melted_df["solo_pin"].notna())]
 
+    # Sometimes the incoming permits have duplicate PINs across PIN
+    # columns, e.g. pin3 == pin4 and both are not null. Make sure to catch
+    # this edge case and remove the duplicate PINs from the set, giving
+    # priority to PINs with lower pin_types (e.g. if pin2 and pin3 share the
+    # same PIN, pin2 should take priority)
+    melted_df = melted_df.sort_values(by=["permit_", "solo_pin", "pin_type"]).drop_duplicates(subset=["permit_", "solo_pin"], keep="first")
```

Comment:
`permit_` in this case is the City of Chicago permit number, and `solo_pin` is the PIN that has been extracted from a column like `pin1` or `pin2`, so this filter ensures that all rows are unique by `(permit_, solo_pin)`.

I normally don't like such long lines in a Python file, but this is currently the style that `chicago/permit_cleaning.py` is written with, and I figure it's better to match that style and reformat everything all at once in a future PR rather than introduce inconsistent styling now.